src/shims/fs.rs

```diff
@@ -730,3 +995,39 @@ fn stat_macos_write_buf<'tcx, 'mir>(
 
     Ok(0)
 }
+
+fn file_type_to_d_type<'tcx, 'mir>(this: &mut MiriEvalContext<'mir, 'tcx>, file_type: std::io::Result<FileType>) -> InterpResult<'tcx, i32> {
+    match file_type {
+        Ok(file_type) => {
+            if file_type.is_dir() {
+                Ok(this.eval_libc("DT_DIR")?.to_u8()? as i32)
+            } else if file_type.is_file() {
+                Ok(this.eval_libc("DT_REG")?.to_u8()? as i32)
+            } else if file_type.is_symlink() {
+                Ok(this.eval_libc("DT_LNK")?.to_u8()? as i32)
+            } else {
+                #[cfg(unix)]
```

Comment:
This makes sense, but please explain why the unix special casing is needed. (This applies to all your `cfg`.)